includes/constants/class-order-statuses.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Class Order_Statuses
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Constants;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+use MyCLabs\Enum\Enum;
+
+/**
+ * This class gives a list of all the possible order status constants.
+ *
+ * @psalm-immutable
+ */
+class Order_Statuses extends Enum {
+	const STATUS_CANCELLED  = 'cancelled';
+	const STATUS_COMPLETED  = 'completed';
+	const STATUS_FAILED     = 'failed';
+	const STATUS_ON_HOLD    = 'on-hold';
+	const STATUS_PENDING    = 'pending';
+	const STATUS_PROCESSING = 'processing';
+	const STATUS_REFUNDED   = 'refunded';
```

Comment:
WDYT about removing `STATUS_` from the constants' names?

Reply:
That actually makes sense since in our actual usage it reads as `Order_Statuses::STATUS_ON_HOLD`.  Avoiding repetition of the word - Status certainly makes sense. I will amend this.

Reply:
Updated via these

802a8a4
67a1dc2
9a93972

Thank you.